src/shared/Microsoft.Git.CredentialManager/Authentication/MicrosoftAuthentication.cs

```diff
@@ -58,14 +178,76 @@ private string FindHelperExecutablePath()
             }
 
             string executableDirectory = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
-            string path = Path.Combine(executableDirectory, helperName);
-            if (!Context.FileSystem.FileExists(path))
+            path = Path.Combine(executableDirectory, helperName);
+            return Context.FileSystem.FileExists(path);
+        }
+
+        private async Task RegisterVisualStudioTokenCacheAsync(IPublicClientApplication app)
+        {
+            Context.Trace.WriteLine("Configuring Visual Studio token cache...");
+
+            // We currently only support Visual Studio on Windows
+            if (PlatformUtils.IsWindows())
             {
-                // We expect to have a helper on Windows and Mac
-                throw new Exception($"Cannot find required helper '{helperName}' in '{executableDirectory}'");
+                // The Visual Studio MSAL cache is located at "%LocalAppData%\.IdentityService\msal.cache" on Windows.
+                // We use the MSAL extension library to provide us consistent cache file access semantics (synchronisation, etc)
+                // as Visual Studio itself follows, as well as other Microsoft developer tools such as the Azure PowerShell CLI.
+                const string cacheFileName = "msal.cache";
```

Comment:
> If there is no msal.cache file succeeds?

The cache is not required to have MSAL get use an access token. Since we're using PATs still we only need an AT once the PAT expires (so we only prompt for re-authentication at that time).

> We also have confirmation msal.cache will live at this location for the fore-seeable future.

Yes. This is the current location for VS2019, and was the location for VS2017 too.

> Is there a scenario where cacheDirectory doesn't exist. What would happen then?

Yes, if Visual Studio 2017+ is not installed. The MSAL extension library we're using to access the token cache [creates it for you](https://github.com/AzureAD/microsoft-authentication-extensions-for-dotnet/blob/c3c9d6ebb2e1467c336e600a5b074f2857bcd46a/src/Microsoft.Identity.Client.Extensions.Msal/MsalCacheStorage.cs#L280-L286).

> Is it at all possible that appData could be null?

No. If the framework returns `null` for the local app data directory, something is seriously wrong with their install/environment that would affect all .NET applications (and maybe even native ones too).

> Just want to double check we have coverage on all sorts of crazy scenarios

❤️ 😁

Reply:
I should also point out that if VS is not installed, and we create and populate the cache with GCM, and then later VS installs, that's a _desirable_ scenario as we've just pre-populated the token cache for them.